test/common/quic/envoy_quic_h3_fuzz.proto

```diff
@@ -219,14 +219,7 @@ message QuicFrame {
   }
 }
 
-message QuicFrameOrJunk {
```

Comment:
Sorry, what do you mean by "re-implementing validation in the Fuzzer"? The junk packet will not use QuicFramer at all and the Envoy code should be resilient to random input, right?

Reply:
The QuicDispatcher made sure that only the correct connection ids are passed to the corresponding QuicConnection/QuicSession. DCHECKs (which are enabled in the fuzzers) verify this in the QuicConnection
by checking `QuicConnection::perspective_`, but there may be other DCHECKs. This will immediately happen when running the fuzzer. Hence, the fuzzer must make sure to avoid any and all DCHECKs in the quiche code base regarding malformed QUIC packets or, in other words, make sure to only generate valid QUIC packets as defined by the (unknown to me) set of DCHECKs. This approach has many downsides: The fuzzer is being constrained and maybe even wrongfully so. It introduces a maintenance burden to always make sure the fuzzer can not generate a QUIC packet that is caught by DCHECKs (which may be introduced in the future). Any mistake made here will introduce noise.

Reply:
To give an example, the fuzzer will immediately trigger the DCHECK in the validation logic of QuicConnection::OnAuthenticatedPublicHeader, quic_connection.cc:1066. This is also the reason why I opted to use the QuicDispatcher in the first place, but it is a bit wasteful.

Reply:
I re-enabled randomized QUIC packets using a valid header.